mono/metadata/assembly.c

```diff
@@ -1222,9 +1227,21 @@ assemblyref_public_tok_checked (MonoImage *image, guint32 key_index, guint32 fla
  * invoked.
  */
 void
-mono_assembly_addref (MonoAssembly *assembly)
+mono_assembly_addref (MonoAssembly *assembly, gboolean pinning)
```

Comment:
Replace the gboolean with an enum or create a pair of functions (addref_pin / addref_nopin). This gboolean is not informative.